src/aosm/azext_aosm/generate_nfd/cnf_nfd_generator.py

```diff
@@ -764,7 +764,14 @@ def _replace_values_with_deploy_params(
                         final_values_mapping_dict[k].append(
                             self._replace_values_with_deploy_params(item, param_name)
                         )
-                    elif isinstance(item, (str, int, bool)) or not v:
+                    elif isinstance(item, (str, int, bool)) or not item:
```

Comment:
Not sure why we have the `or not item` it here but not on line 738.  I suspect that it is line 738 that is wrong.

Reply:
There is another branch for the None case level with line 738, but I can't see why both blocks don't have the same structure. I'll change 738..